migration to v10

guewen opened the pull request:
This is the one-to-one migration of the connector. More ambitious refactorings will follow in other pull requests (#212, ..).

Commit:
Make the connector addon installable

Commit:
Replace 'select=True' by 'index=True'

Commit:
Rename openerp to odoo in Python files

Commit:
Rename xml files with odoo instead of openerp

Commit:
Rename openerp to odoo in docs

Commit:
Replace monkey-patch by a proper inherit

Now that we have a root model 'base', we can properly override the
methods to fire the events.

Commit:
Remove usage of @api.one

Commit:
Replace ConnectorSession by odoo.Environment

Commit:
Upgrade version number

Commit:
Merge JobStorage into Job

The reason of being for the separation was to be able to support another
storage than postgres if needed.  Nothing has ever been done in this
way and the current backend storage (postgres) has proved to be
efficient.

Commit:
Extract the queue_job from connector

lasley commented on connector/backend_model.py:
```diff
@@ -19,7 +19,7 @@
 #
 ##############################################################################
```
Please use new, shorter header

lasley commented on connector/connector.py:
```diff
@@ -351,14 +349,14 @@ class Binder(ConnectorUnit):
-    def to_openerp(self, external_id, unwrap=False):
-        """ Give the OpenERP ID for an external ID
+    def to_odoo(self, external_id, unwrap=False):
+        """ Give the ODoo ID for an external ID
```
s/ODoo/Odoo

lasley commented on connector/doc/index.rst:
```diff
@@ -51,7 +51,7 @@ Connectors based on the framework
 * `Prestashop Connector <https://github.com/OCA/connector-prestashop>`_
 * `solerp (Solr Connector) <https://github.com/akretion/solerp>`_
-* `Odoo Multi Company <http://www.openerp.net.cn/new-intercompany-process-module/>`_
+* `Odoo Multi Company <http://www.odoo.net.cn/new-intercompany-process-module/>`_
```
This URI is (and previously was) invalid. Is this it - https://github.com/OCA/connector-odoo2odoo ?

lmignon replied:
@guewen Can you remove this URI. Indeed it's invalid.

Commit:
Remove ConnectorSession

Commit:
Typo

Commit:
Remove migration code for a previous odoo version

Commit:
Change travis version to 10.0

lasley commented on connector/queue/queue_data.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
+<odoo>
```
`<odoo noupdate="1">` & can remove the data tag

guewen replied:
I had to leave it because there is one part with `noupdate="1"` and another one with `noupdate="0"` in the file.

lasley reviewed (approved):
Minor nitpicks, code no test 👍 

Side note - the producer is very sexy now w/ the use of the BaseModel

lasley commented:
FYI functional test on this looks good too

guewen commented:
This PR still includes the queue, we'll have to continue the work on #212 which was already split in `connector` and `queue_job` (but the latter has to be dropped as moved in its own repo).

lasley commented:
Thanks for clarifying @guewen - I was kind of curious whether we would be refactoring for the new queue.

It seems to me like we should still continue forward with the merge of this one though, as it works in its entirety & we are starting to receive other PRs doing the same upgrade. 

Stripping out the queue/jobs should be fairly transparent in terms of the external API, or am I incorrect?

lmignon commented:
IMO we should wait for the move of the queue to it's own repo since this change is not transparent in terms of the external API.

lasley commented:
Thanks for the clarification @lmignon

bealdav commented:
HI Guewen, 
Your are not afraid by a new rename ?
`s/odoo_id/erp_id` can save you for eternity ;-)

guewen commented:
@bealdav that's an ERP they say.

Anyway, no, I'm not afraid by a new rename, odoo is spread all over the application (xml, python files, ...) so if it had to change again, renaming a few more occurrences would not be so much work.

gurneyalex commented:
Travis is read because of https://github.com/OCA/odoo-sphinx-autodoc/issues/6 I think

gurneyalex commented:
OCA/odoo-sphinx-autodoc#7 should fix the build.

guewen commented:
This PR still need some work, it still contains the queue job that has now been extracted to https://github.com/OCA/queue

Commit:
Use short license headers

Commit:
Add is_module_installed function

It has been removed from queue_job, move it back here

Commit:
Rework slighty the default Binder

* Remove notion of "id", we work with recordsets.
* Rename 'to_backend' to 'to_external' to avoid confusion with the
  notion of backend
* Rename 'to_odoo' to 'to_internal' for reciprocity
* Rename m2o_to_backend and backend_to_m2o in the same manner

Commit:
Add a test_connector addon

So we can write tests using real pieces of the framework on real models
instead of using mocks which is quite limited and wring things in
confusing ways.

* Fix the related action implementation
* Move the tests of the Default Binding in this module so we have tests
  that reflect the reality

Commit:
Add oca_dependencies.txt

Commit:
Start to actualize documentation

guewen commented:
FYI tests are green travis is red only because of the docs build, which should be better once we have a new release of odoo-sphinx-autodoc (https://github.com/OCA/odoo-sphinx-autodoc/pull/7).

I'll now battle-test a little with a real implementation.

Docs and examples will need to be updated but that can wait.

lmignon commented:
Thank you @guewen for all this work!

lmignon commented on connector/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+          'res_partner_view.xml',
+          ],
+ 'installable': True,
+ 'application': True,
```
IMO it's not really an application since it doesn't provide functionality alone...

guewen replied:
right, will change that

guewen commented:
I should also take the last commits from 9.0

Commit:
Remove 'env' argument from ConnectorEnvironment

We probably want to use the same env that we have in the backend_record!
Besides, having 2 environments bring issues when we modify one but not
the other one.

guewen commented:
In the last commit, I remove the `env` of the `ConnectorUnit` as I think we should use the one of the backend_record.

Commit:
Remove application flag, not an application

yvaucher commented on connector/doc/guides/bootstrap_connector.rst:
```diff
@@ -172,22 +172,20 @@ Environment
-                return env
+      @contextmanager
+      @api.multi
+      def get_environment(self, model_name):
```
indentation is 6 instead of 8

guewen replied:
Doc is outdated and need to be refreshed but not in this PR.

guewen replied:
And: yes we could add `get_environment` in the abstract `connector.backend`. Yes the `model_name` is required for the `ConnectorEnvironment`.

LeartS commented:
Hi guys, what's the status of this PR? Can it be merged? I'm using the connector module in this PR to build a connector with a SaaS search engine and it seems to work perfectly.

LeartS reviewed (approved):
👍 Quick code review + used to build a specific connector module on top of it, seems to work great.

leemannd committed:
str to ustr

Commit:
Install odoo-sphinx-autodoc from source

Because it includes a fix for Odoo 10.0 which is not released yet.

Commit:
Tentative to build docs

guewen commented:
Docs build is now green as well, ready to be merged.

lmignon reviewed (approved):
@guewen Thank you for all this work! Can you remove the invalid URI into connector/doc/inddex.rst? Once done I'll merge this PR

Commit:
Remove invalid URL

guewen commented:
@lmignon done

lmignon commented:
@guewen  thank you
